Cargo.toml

```diff
@@ -29,7 +29,7 @@ tokio = { version = "1.39", optional = true, features = ["macros"] }
 strum = { version = "0.26", features = ["derive"] }
 trait-set = "0.3"
 leptos-use = "0.11"
-si_format = "0.1"
+format_num = "0.1.0"
```

Comment:
This library seems to get the job done. My only concern is that it hasn't been touched in over 4 years and was only made in 3 commits. At that point, I don't see the benefit to using this library when the hand-rolled implementation was less complex (we only use 1 pattern and this library uses regex to parse the pattern), and the hand-rolled implementation was thoroughly tested to make sure it formatted numbers in the way we expect it to.

Reply:
Yeah, doesn't look like the best library. I'll mark this as draft, until we find (or make) a better crate. 

This looks like a good simple possibility, I'll work on it soon: https://stackoverflow.com/a/23794345/1655478

Reply:
[There's always the function from the old PR.](https://github.com/LemmyNet/lemmy-ui-leptos/blob/5b2e653467ee9ee34898b03de15b329a15faabb6/src/utils/format_number_si.rs) Not sure if it makes sense to extract into its own crate, but it's lightweight and behaves the way we expect it to.

Reply:
Ya I think it would be useful to extract that to its own crate. LMK if you want to, otherwise I can. Maybe `pretty_si_number_format` or something.

Reply:
Also do you remember where that algorithm came from?

Reply:
I think it's fine to leave the implementation in this codebase for now. When I get some free time where I'm not working on this UI or the inferno one, I'll play around with adding some options to the algorithm. I think it's a bit too specific to our usecase to be extracted to its own crate for the time being.

As to where the algorithm came from, I came up with it myself.

Reply:
Mmmk, in that case I'll close this so you can just re-add that last one.

I'd pry encourage you to make the crate sooner rather than later, even if it only fits our one use-case, so others could potentially start contributing and adding the options they want themselves.